tensorflow/core/common_runtime/dml/dml_device_factory.cc

```diff
@@ -25,6 +25,32 @@ using Microsoft::WRL::ComPtr;
 
 namespace tensorflow {
 
+// Constructs a physical_device_desc string from an adapter. DML's
+// physical_device_desc is JSON-encoded which allows for easier parsing.
+//
+// Example CUDA string:
+//   'device: 0, name: TITAN V, pci bus id: 0000:65:00.0, compute
+//   capability: 7.0'
+//
+// Example DML string:
+//   '{"name": "NVIDIA TITAN V", "vendor_id": 4318, "device_id": 7553,
+//   "driver_version": "27.21.14.5206"}'
+static std::string GetPhysicalDeviceDesc(const DmlAdapter& adapter) {
+  auto driver_ver = adapter.DriverVersion().parts;
+  std::string driver_ver_str = strings::StrCat(
+      driver_ver.a, ".", driver_ver.b, ".", driver_ver.c, ".", driver_ver.d);
+
+  std::stringstream ss;
+  ss << '{';
+  ss << "\"name\": \"" << adapter.Name() << "\", ";
```

Comment:
_Might_ be useful to include the adapter description as well. I recall seeing some instances where the name is actually quite generic but the description is more informative. However, this may have been with co-engineering drivers.